assets/js/clipboard.js

```diff
@@ -0,0 +1,52 @@
+const svgCopy =
+    '<svg aria-hidden="true" height="16" viewBox="0 0 16 16" version="1.1" width="16" data-view-component="true"><path fill-rule="evenodd" d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 010 1.5h-1.5a.25.25 0 00-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 00.25-.25v-1.5a.75.75 0 011.5 0v1.5A1.75 1.75 0 019.25 16h-7.5A1.75 1.75 0 010 14.25v-7.5z"></path><path fill-rule="evenodd" d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0114.25 11h-7.5A1.75 1.75 0 015 9.25v-7.5zm1.75-.25a.25.25 0 00-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 00.25-.25v-7.5a.25.25 0 00-.25-.25h-7.5z"></path></svg>';
+const svgCheck =
+    '<svg aria-hidden="true" height="16" viewBox="0 0 16 16" version="1.1" width="16" data-view-component="true"><path fill-rule="evenodd" fill="rgb(63, 185, 80)" d="M13.78 4.22a.75.75 0 010 1.06l-7.25 7.25a.75.75 0 01-1.06 0L2.22 9.28a.75.75 0 011.06-1.06L6 10.94l6.72-6.72a.75.75 0 011.06 0z"></path></svg>';
+
+
+const addCopyButtons = (clipboard) => {
+    var els = document.getElementsByClassName("highlight");
+    // for each highlight
+    for (var i = 0; i < els.length; i++) {
+        if (els[i].getElementsByClassName("clipboard-button").length) continue;
+
+        // find pre > code inside els[i]
+        let codeBlocks = els[i].getElementsByTagName("code");
+
+         // line numbers are inside first code block
+        let lastCodeBlock = codeBlocks[codeBlocks.length - 1];
+        const button = document.createElement("button");
+        button.className = "clipboard-button";
+        button.type = "button";
+        button.innerHTML = svgCopy;
+        // remove every second newline from lastCodeBlock.innerText
+        button.addEventListener("click", () => {
+            clipboard.writeText(lastCodeBlock.innerText.replace(/\n\n/g, "\n")).then(
+                () => {
+                    button.blur();
+                    button.innerHTML = svgCheck;
+                    setTimeout(() => (button.innerHTML = svgCopy), 2000);
+                },
+                (error) => (button.innerHTML = "Error")
+            );
+        });
+        // find chroma inside els[i]
+        let chroma = els[i].getElementsByClassName("chroma")[0];
+        els[i].insertBefore(button, chroma);
+        console.log(els[i].lastChild)
+    }
+}
+
+function initClipboard() {
+    if (navigator && navigator.clipboard) {
+        addCopyButtons(navigator.clipboard);
+    } else {
+        const script = document.createElement("script");
+        script.src =
+            "https://cdnjs.cloudflare.com/ajax/libs/clipboard-polyfill/2.7.0/clipboard-polyfill.promise.js";
+        script.integrity = "sha256-waClS2re9NUbXRsryKoof+F9qc1gjjIhc2eT7ZbIv94=";
+        script.crossOrigin = "anonymous";
+        script.onload = () => addCopyButtons(clipboard);
+        document.body.appendChild(script);
+    }
+}
```

Comment:
can the script tag be moved to the `head`? can do conditional rendering logic there

Reply:
@jackyzha0 I could propose moving the `initClipboard` function as is to the `head`, though if I understand you correctly, you would want the script optionally included in `html`? 

We might be able to remove this optionally included script entirely as It seems almost all modern browsers ([except for IE](https://caniuse.com/mdn-api_navigator_clipboard)) support the clipboard API. Even the npm package for `clipboard-polyfill` mentions this, and that it will be deprecated.

I'll go ahead and test this on my desktop, and then modify the PR accordingly :)!